ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Serialisation/SerialiseNodeToClient.hs

```diff
@@ -135,6 +261,16 @@ dispatchDecoderErr ccfg version =
 after :: (a -> b -> d -> e) -> (c -> d) -> a -> b -> c -> e
 after f g x y z = f x y (g z)
 
+{-------------------------------------------------------------------------------
+  Ledger Config
+-------------------------------------------------------------------------------}
+
+instance SerialiseHFC xs => SerialiseNodeToClient (HardForkBlock xs) (PerEraLedgerConfig xs) where
+  encodeNodeToClient ccfg version (PerEraLedgerConfig xs)
+    = encodeNodeToClientNP encodeNodeToClient ccfg version xs
+  decodeNodeToClient ccfg version
+    = PerEraLedgerConfig <$> decodeNodeToClientNP decodeNodeToClient ccfg version
```

Comment:
@nfrisby You mentioned that we have an invariant that using `blk ~ A` and using `blk ~ HardForkBlock '[A]` should be compatible. This is sort of broken: the encoding for `LedgerConfig A` and `LedgerConfig HardForkBlock '[A]` will be different as they are different types. We'd have to make an attempt to convert the `LedgerConfig HardForkBlock '[A]` back into a `LedgerConfig A`. Does that seem like the correct thing to do?

Reply:
I think they only have to be "(morally) isomorphic", not necessarily "compatible". IE I'm not sure we would need an `A` node and an `HFC '[A]` node to actually be able to communicate.

I'm guessing there are already other `SerialiseNodeTo*` instances where the `HFC` layer leads to incompatibilities. Please check me on that.

Reply:
Humm yeah. I think you're right here. We do want to be backwards compatible with old clients that use `A` instead of `HFB '[A]`, but such old clients will not know of the new `GetLedgerConfig` query and hence will never request/decode it. So we're actually still compatible in that sense.